### PR TITLE
Rocthrust bringup

### PR DESCRIPTION
**Pull request by iotamudelta**

### CODE REVIEW

**Comment by iotamudelta** [comment]
won't pass CI - feature for ROCm 2.5+ w/ rocthrust integration

File: cmake/Dependencies.cmake [hunk 1]
```diff
@@ -889,13 +890,13 @@ if(USE_ROCM)
     endforeach()
 
     set(Caffe2_HIP_INCLUDE
-      ${hip_INCLUDE_DIRS} ${hcc_INCLUDE_DIRS} ${hsa_INCLUDE_DIRS} ${rocrand_INCLUDE_DIRS} ${hiprand_INCLUDE_DIRS} ${rocblas_INCLUDE_DIRS} ${miopen_INCLUDE_DIRS} ${thrust_INCLUDE_DIRS} $<INSTALL_INTERFACE:include> ${Caffe2_HIP_INCLUDE})
+      ${thrust_INCLUDE_DIRS} ${hipcub_INCLUDE_DIRS} ${rocprim_INCLUDE_DIRS} ${miopen_INCLUDE_DIRS} ${rocblas_INCLUDE_DIRS} ${hiprand_INCLUDE_DIRS} ${rocrand_INCLUDE_DIRS} ${hip_INCLUDE_DIRS} ${hcc_INCLUDE_DIRS} ${hsa_INCLUDE_DIRS} $<INSTALL_INTERFACE:include> ${Caffe2_HIP_INCLUDE})
```
Comment (saadrahim) [inline cmake/Dependencies.cmake]: This is after the fact but shouldn't ${thrust_INCLUDE_DIRS}  be rocthrust_INCLUDE_DIRS?

File: cmake/Dependencies.cmake [hunk 2]
```diff
@@ -889,13 +890,13 @@ if(USE_ROCM)
...
 
     # This is needed for library added by hip_add_library (same for hip_add_executable)
     hip_include_directories(${Caffe2_HIP_INCLUDE})
 
     set(Caffe2_HIP_DEPENDENCY_LIBS
-      ${rocrand_LIBRARIES} ${hiprand_LIBRARIES} ${hipsparse_LIBRARIES} ${PYTORCH_HIP_HCC_LIBRARIES} ${PYTORCH_MIOPEN_LIBRARIES})
+      ${rocrand_LIBRARIES} ${hiprand_LIBRARIES} ${hipsparse_LIBRARIES} ${PYTORCH_HIP_HCC_LIBRARIES} ${PYTORCH_MIOPEN_LIBRARIES} ${hipcub_LIBRARIES})
```
Comment (saadrahim) [inline cmake/Dependencies.cmake]: should ${rocthrust_LIBRARIES} be added to this line?

File: cmake/public/LoadHIP.cmake [hunk 3]
```diff
@@ -94,6 +94,27 @@ ELSE()
   SET(MIOPEN_PATH $ENV{MIOPEN_PATH})
 ENDIF()
 
+# ROCPRIM_PATH
```
Comment (saadrahim) [inline cmake/public/LoadHIP.cmake]: Wondering if this section should come from code like below:

# rocPRIM
--
2 |  
3 | The rocPRIM is a header-only library providing HIP parallel primitives for developing
… |  
74 | target_link_libraries(<your_target> roc::rocprim_hip)

File: cmake/Dependencies.cmake (hunk 2 above)
Comment (iotamudelta) [reply cmake/Dependencies.cmake]: there are no rocThrust library objects as far as I know?

File: cmake/Dependencies.cmake (hunk 1 above)
Comment (iotamudelta) [reply cmake/Dependencies.cmake]: we need thrust_INCLUDE_DIRS (they've been overloaded to also mean something else...).

File: cmake/public/LoadHIP.cmake (hunk 3 above)
Comment (iotamudelta) [reply cmake/public/LoadHIP.cmake]: not clear what you mean? we link the rocprim libraries cmake returns.

File: cmake/Dependencies.cmake (hunk 2 above)
Comment (saadrahim) [reply cmake/Dependencies.cmake]: Please take a look at Paul's comments in https://github.com/RadeonOpenCompute/rocm-cmake/pull/9

Comment (saadrahim) [reply cmake/Dependencies.cmake]: I am very new to cmake, so some of these questions may be very basic.

Comment (saadrahim) [reply cmake/Dependencies.cmake]: I am trying to understand building process better for http://ontrack-internal.amd.com/browse/SWDEV-195246

Comment (pfultz2) [reply cmake/Dependencies.cmake]: @iotamudelta The cmake target is for usage requirements not just for library objects.

**Pull request merged**